Automattic/wp-calypso
# Edge Cache: Add edge cache toggle button

Related to https://github.com/Automattic/dotcom-forge/issues/2769
Fixes https://github.com/Automattic/dotcom-forge/issues/2770

## Proposed Changes
This pr proposes adding a checkbox in order to enable and disable the edge cache.
It also introduces clearing edge cache as well, in case it's enabled.

## Testing Instructions


### Test functionality
1. Apply the diff D113979-code
2. Create a site and make it atomic.
3. Visit blog RC and ensure that:
    - When the cache is enabled, The label "ENABLED" next to Edge Cache: is appearing on the top of the page. Also, ensure that the sticker edge-cache-enabled is added to the site
    - When the cache is disabled, The label "DISABLED" next to Edge Cache: is appearing on the top of the page. Also, ensure that the sticker edge-cache-disabled is added to the site

### Test UI

1. Visit `Settings > Hosting configuration` and find the `Cache section
2. Ensure the UI looks like the following

| Enabled      | Disabled |
| :---:      |    :----:   |  
| ![edge_cache_enabled](https://github.com/Automattic/wp-calypso/assets/497103/53bbcb00-e389-4f1c-966a-88d45e023385)| ![edge_cache_disabled](https://github.com/Automattic/wp-calypso/assets/497103/7eec5531-d4f8-4134-9685-fcbf2912650a)|

### Test loader

1. Visit `Settings > Hosting configuration` and find the `Cache section
2. Ensure the UI looks like the following on first load
![edge_cache_placeholder](https://github.com/Automattic/wp-calypso/assets/497103/59396411-0ef3-4773-aa2e-99a1bae00b74)





## Pre-merge Checklist
